### PR TITLE
feat(pipe): use jsdom everywhere

### PR DESCRIPTION
**Pull request by tripodsan**
see #337, see #336 

## Improvements
- remove the use of `htast` and `hast` where possible
- the VDOM transformer now directly transforms the `mdast` to `hast` to `dom`. before it used to create the jsdom via a html-string roundtrip.

## BREAKING CHANGES:
- remove `htast` from `context.content` schema
- remove `hast` from `context.response` schema
- add `document` from to `context.response` schema
- `context.htast` and `context.response.hast` is no longer generated
- mdast to jsdom transformation handlers can only return hast (not string, not vdom)
- html postprocessing pipeline step was renamed from `hast` to `post`
- mixing inline html and markdown no longer works (eg: `<em>This is **strong**</em>`)

## Further Work

I think that we should generate the jsdom directly from the `mdast`. this would even speed up the transformation more. currently, there is no utility available for this. we would need to copy major parts of [mdast-util-to-hast](https://github.com/syntax-tree/mdast-util-to-hast).

### CODE REVIEW

File: src/defaults/html.pipe.js [hunk 1]
```diff
@@ -63,18 +63,20 @@ const htmlpipe = (cont, payload, action) => {
     .before(selectstrain)
     .before(selecttest)
     .before(html).expose('html')
-    .before(responsive)
+    // todo: responsive used to operate on the htast, therefore ignored if content.document was used
+    // todo: there is similar logic in the image-handler during jsdom creation....
+    // .before(responsive)
```
Comment (tripodsan) [inline src/defaults/html.pipe.js]: @trieloff WDYT?

File: src/utils/hast-util-to-dom.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ * INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+ * OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+ * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+ * THIS SOFTWARE.
+ */
+
+// This was copied from https://github.com/syntax-tree/hast-util-to-dom/blob/master/src/index.js
```
Comment (tripodsan) [inline src/utils/hast-util-to-dom.js]: see https://github.com/syntax-tree/hast-util-to-dom/issues/3

**Comment by ramboz** [comment]
@tripodsan These are triggered when you use custom HTML elements and attributes. So Something like:
```
<my-tag has-some-attribute/>
```

I have a test for those in https://github.com/adobe/helix-pipeline/blob/051375678c3a6280e437e89dc693559b25fae2e9/test/testHTMLFromMarkdown.js#L304-L316, so you can probably leverage something similar

**Comment by tripodsan** [comment]
@ramboz when I run your test, it never gets into this function. I don't know what changed...but the test succeeds.

**Comment by ramboz** [comment]
@tripodsan Alright. I'll review this tomorrow

**Pull request merged**